infra/embedding_step_functions/unified_embedding/utils/tracing.py

```
@@ -0,0 +1,388 @@
+"""X-Ray tracing utilities for Lambda functions."""
+
+import os
+from contextlib import contextmanager
+from typing import Optional, Dict, Any
+from functools import wraps
+
+try:
+    from aws_xray_sdk.core import xray_recorder, patch_all
+    from aws_xray_sdk.core.context import Context
```

Ruff (0.12.2)

`aws_xray_sdk.core.context.Context` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)